crates/hir-ty/src/tests/macros.rs

```diff
@@ -745,7 +745,7 @@ fn infer_builtin_macros_concat() {
         expect![[r#"
             !0..13 '"helloworld!"': &'static str
             65..121 '{     ...")); }': ()
-            75..76 'x': &'static str
+            75..76 'x': &'? str
```

Comment:
This is another theme repeating in the tests: we lost almost all of our lifetimes. Due to the significance of this I checked this, and this is kinda intentional: rustc has a region check pass (well, not exactly a pass) (https://rustc-dev-guide.rust-lang.org/type-inference.html#region-constraints) and without it our regions won't be resolved. We could add that pass later, or we could instead wait until we properly support lifetimes.